eng/scripts/get-aspire-cli.sh

```diff
@@ -0,0 +1,438 @@
+            return 1
+        fi
+
+        if ! unzip -o "$archive_file" -d "$destination_path"; then
+            printf "Error: Failed to extract ZIP archive: %s\n" "$archive_file" >&2
+            return 1
+        fi
+    else
+        # Use tar for tar.gz files on Unix systems
+        if ! command -v tar >/dev/null 2>&1; then
+            printf "Error: tar command not found. Please install tar to extract tar.gz files.\n" >&2
+            return 1
+        fi
+
+        if ! tar -xzf "$archive_file" -C "$destination_path"; then
+            printf "Error: Failed to extract tar.gz archive: %s\n" "$archive_file" >&2
+            return 1
+        fi
+    fi
+
+    say_verbose "Successfully unpacked archive"
+}
+
+# Main script
+main() {
+    local os arch runtimeIdentifier url filename checksum_url checksum_filename
+    local cli_exe cli_path
+
+    # Parse command line arguments
+    parse_args "$@"
+
+    # Show help if requested
+    if [[ "$SHOW_HELP" == true ]]; then
+        show_help
+        exit 0
+    fi
+
+    # Set default OutputPath if empty
+    if [[ -z "$OUTPUT_PATH" ]]; then
+        OUTPUT_PATH="$(pwd)/aspire-cli"
+    fi
+
+    # Create a temporary directory for downloads
+    local temp_dir
+    temp_dir=$(mktemp -d -t aspire-cli-download-XXXXXXXX)
+
+    if [[ ! -d "$temp_dir" ]]; then
+        say_verbose "Creating temporary directory: $temp_dir"
+        if ! mkdir -p "$temp_dir"; then
+            printf "Error: Failed to create temporary directory: %s\n" "$temp_dir" >&2
+            return 1
+        fi
+    fi
+
+    # Cleanup function for temporary directory
+    cleanup() {
+        if [[ -n "${temp_dir:-}" ]] && [[ -d "$temp_dir" ]]; then
+            if [[ "$KEEP_ARCHIVE" != true ]]; then
+                say_verbose "Cleaning up temporary files..."
+                rm -rf "$temp_dir" || printf "Warning: Failed to clean up temporary directory: %s\n" "$temp_dir" >&2
+            else
+                printf "Archive files kept in: %s\n" "$temp_dir"
+            fi
+        fi
+    }
+
+    # Set trap for cleanup on exit
+    trap cleanup EXIT
+
+    # Detect OS and architecture if not provided
+    local detected_os detected_arch
+
+    if [[ -z "$OS" ]]; then
+        if ! detected_os=$(get_os); then
+            printf "Error: Unsupported operating system. Current platform: %s\n" "$(uname -s)" >&2
+            return 1
+        fi
+        os="$detected_os"
+    else
+        os="$OS"
+    fi
+
+    if [[ -z "$ARCH" ]]; then
+        if ! detected_arch=$(get_arch); then
+            printf "Error: Unsupported architecture. Current architecture: %s\n" "$(uname -m)" >&2
+            return 1
+        fi
+        arch="$detected_arch"
+    else
+        arch="$ARCH"
+    fi
+
+    # Construct the runtime identifier
+    runtimeIdentifier="${os}-${arch}"
+
+    # Determine file extension based on OS
+    if [[ "$os" == "win" ]]; then
+        extension="zip"
+    else
+        extension="tar.gz"
+    fi
```

Comment:
I think the `osx` files are extension `.zip`.

https://github.com/dotnet/aspire/blob/853c9189814850a686f5445b9b1ac0f5a76e3b30/eng/clipack/Common.projitems#L6-L7